packages/@aws-cdk/aws-sagemaker/lib/model.ts

```diff
@@ -0,0 +1,405 @@
+   * @param scope the Construct scope.
+   * @param id the resource id.
+   * @param attrs the attributes of the model to import.
+   */
+  public static fromModelAttributes(scope: Construct, id: string, attrs: ModelAttributes): IModel {
+    const modelName = attrs.modelName;
+    const role = attrs.role;
+
+    class Import extends ModelBase {
+      public readonly modelName = modelName;
+      public readonly role = role;
+      public readonly grantPrincipal: iam.IPrincipal;
+      public readonly modelArn: string;
+
+      constructor(s: Construct, i: string) {
+        super(s, i);
+
+        this.modelArn = cdk.Stack.of(this).formatArn({
+          service: 'sagemaker',
+          resource: 'model',
+          resourceName: this.modelName,
+        });
+        this.grantPrincipal = role || new iam.UnknownPrincipal({ resource: this });
+        if (attrs.securityGroups) {
+          this._connections = new ec2.Connections({
+            securityGroups: attrs.securityGroups,
+          });
+        }
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Returns the ARN of this model.
+   * @attribute
+   */
+  public readonly modelArn: string;
+  /**
+   * Returns the name of the model.
+   * @attribute
+   */
+  public readonly modelName: string;
+  /**
+   * Execution role for SageMaker Model
+   */
+  public readonly role?: iam.IRole;
+  /**
+   * The principal this Model is running as
+   */
+  public readonly grantPrincipal: iam.IPrincipal;
+  private readonly subnets: ec2.SelectedSubnets | undefined;
+  private readonly containers: CfnModel.ContainerDefinitionProperty[] = [];
+
+  constructor(scope: Construct, id: string, props: ModelProps = {}) {
+    super(scope, id, {
+      physicalName: props.modelName,
+    });
+
+    this._connections = this.configureNetworking(props);
+    this.subnets = (props.vpc) ? props.vpc.selectSubnets(props.vpcSubnets) : undefined;
+
+    // set the sagemaker role or create new one
+    this.role = props.role || this.createSageMakerRole();
+    this.grantPrincipal = this.role;
+
+    (props.containers || []).map(c => this.addContainer(c));
+
+    // apply a name tag to the model resource
+    cdk.Tags.of(this).add(NAME_TAG, this.node.path);
+
+    const model = new CfnModel(this, 'Model', {
+      executionRoleArn: this.role.roleArn,
+      modelName: this.physicalName,
+      primaryContainer: cdk.Lazy.any({ produce: () => this.renderPrimaryContainer() }),
+      vpcConfig: cdk.Lazy.any({ produce: () => this.renderVpcConfig() }),
+      containers: cdk.Lazy.any({ produce: () => this.renderContainers() }),
+    });
+    this.modelName = this.getResourceNameAttribute(model.attrModelName);
+    this.modelArn = this.getResourceArnAttribute(model.ref, {
+      service: 'sagemaker',
+      resource: 'model',
+      resourceName: this.physicalName,
+    });
+
+    // post-construction validation
+    this.node.addValidation({ validate: () => this.validateContainers() });
```

Comment:
why is this not called inside of `this.renderContainers()`?

Reply:
I suppose it just seemed intuitive to me to perform synthesis-time validation using the validation hook. I can certainly move the check though if you'd like.

Reply:
I think we can still have `this.validateContainers()`, but that should be called istelf inside `this.renderContainers()`. I call this out because it seems different from the designs of other L2s, and for maintainability purposes, it's probably good to keep things relatively in line with other L2s.